SimDataFormats/GeneratorProducts/interface/LHEEventProductLite.h

```diff
@@ -0,0 +1,124 @@
+#ifndef SimDataFormats_GeneratorProducts_LHEEventProductLite_h
+#define SimDataFormats_GeneratorProducts_LHEEventProductLite_h
+
+#include <memory>
+#include <vector>
+#include <string>
+
+#include "SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h"
+#include "SimDataFormats/GeneratorProducts/interface/LesHouches.h"
+#include "SimDataFormats/GeneratorProducts/interface/PdfInfo.h"
+#include "SimDataFormats/GeneratorProducts/interface/WeightsInfo.h"
+
+class LHEEventProductLite {
+    public:
+    typedef gen::PdfInfo PDF;
+    typedef float WGT;
+
+    void copyWeightsVector (std::vector<float> & outV,
+                            const std::vector<gen::WeightsInfo> & inV)
+      {
+        for (unsigned int i = 0 ; i < inV.size () ; ++i)
+        	outV.push_back (inV.at (i).wgt) ;
+        return ;
+      }
+
+    LHEEventProductLite() {}
+        LHEEventProductLite(const lhef::HEPEUP hepeup) : 
+      hepeup_(hepeup), originalXWGTUP_(0) {}
+    LHEEventProductLite(const lhef::HEPEUP &hepeup,
+            const double originalXWGTUP) : 
+      hepeup_(hepeup), originalXWGTUP_(originalXWGTUP) {}
+
+    LHEEventProductLite(const LHEEventProduct * lep)
+      {
+        hepeup_         = lep->hepeup () ;
+        originalXWGTUP_ = lep->originalXWGTUP () ;
+        npLO_           = lep->npLO () ;
+        npNLO_          = lep->npNLO () ;
+
+        if (lep->pdf ())
+          {
+            this->setPDF (*lep->pdf ()) ;
+          }
+        pdf_.reset(new PDF()) ; 
```

Comment:
why do you first setPDF and then immediately afterwards reset it to something dummy?